src/Table/DetailTable.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import styled from 'styled-components';
+
+const DetailTableWrapper = styled.div`
+    width: 100%;
+    table {
+        width: 100%;
+        .key {
+            font-style: italic;
+        }
+    }
+`;
+
+type Data = [string, any];
+
+interface Props {
+    data: Data[];
+}
+
+export class DetailTable extends React.PureComponent<Props> {
+
+    constructor(props: Props) {
+        super(props);
+    }
+
+    getRows(data: Data[]) {
+        return data.map(d => {
+            switch (typeof (d[1])) {
+                case 'object':
+                    if (d[1] === null) return this.renderRows(d[0], 'NULL');
+                    return this.renderRows(d[0], JSON.stringify(d[1]));
+                case 'number':
+                    return this.renderRows(d[0], (d[1] as number).toLocaleString());
+                case 'string':
+                    return this.renderRows(d[0], d[1]);
+                default:
+                    if (d[1] === undefined) return;
+                    return this.renderRows(d[0], d[1]);
+            }
+        });
+    }
```

Comment:
Better use User-Defined Type Guards, https://www.typescriptlang.org/docs/handbook/advanced-types.html

Reply:
For more info, I don't think we should try to get rid of type cast (`as number`) here. Not sure typescript can refine type on this or not, or else we can use function type guards for all type instead
```ts
function isNumber(value: any): value is number {
    return typeof value === 'number'
}
```

Reply:
But in this case I use `as number` for `toLocaleString` which is from lib.